File: 0121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        maxStock = 0
        minValue = sys.maxsize
        
        
        for i in range(0, len(prices)):
            if(prices[i] < minValue):
                minValue = prices[i]
            else:
                maxStock = max(maxStock, prices[i] - minValue)
        
        return maxStock
        

File: 0121.cpp
class Solution {
public:
    int low;
    int high;
    
    int maxProfit(vector<int>& prices) {
        low = prices[0];
        high = 0;
        
        for(int i=0; i<prices.size(); i++) {

            low = min(prices[i], low);
            high = max(prices[i] - low, high);
        }
        
        return high;
    }
};